### PR TITLE
Isolate prep period

### PR DESCRIPTION
**Pull request by Sybrand**
- [x] Stop using request object for prep period.
- [x] Get rid of save button.
- [x] Write / update front end tests.
- [x] Write / update back-end test. (I think I simplified the tests somewhat - removed a lot of now redundant tests)
- [x] Split HTML part from PDF gen to allow for unit test on HTML portion.
# Test Links:
[Percentile Calculator](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1856.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

**Review by conbrad — APPROVED** [review]
I like all the red deleted code! It seems we have different JSON file formatting settings.

File: api/app/tests/hfi/test_pdf_generator.py [hunk 1]
```diff
@@ -26,6 +45,8 @@ def test_generate_pdf():
                                                datetime.fromisocalendar(2022, 2, 2), jinja_env)
         assert len(pdf_bytes) > 0
         assert pdf_filename is not None
+        # Would be nice to check if the pdf hash has changed, but the pdf generated by pdfkit is non
+        # deterministic.
```
[the remark this answers is not in the document]
Comment (Sybrand) [reply api/app/tests/hfi/test_pdf_generator.py:50]: It's not though - the generation date is being passed in in the test - so it's something else. (Unless I misread the code)

**Review by andrea-williams — APPROVED** [review]
Given the time constraints before review, I didn't look at this closely. I'm sure there are spelng misstkaes everywhere. Que sera sera.

File: api/app/tests/hfi/test_pdf_generator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (conbrad) [reply api/app/tests/hfi/test_pdf_generator.py:50]: Nope, you're correct

**Pull request merged**